src/js/components/sorter/Sorter.jsx

```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+import { ButtonToolbar, ToggleButtonGroup, ToggleButton } from 'react-bootstrap';
+import './sorter.style.less';
+
+class Sorter extends Component {
+
+  state = {
+    sortby: this.props.sortby
+  };
+
+  clickHandler = () => {
+    if (event.target.value) {
+      this.setState({sortby: event.target.value});
+      this.props.changeHandler(this.state);
```

Comment:
setState is async function so you need to be 100%  sure that state is already set when you are attempting to use it right after it's set. To ensure this setState provide a callback function that is triggered right after setting the state.

As i metioned async nature of setState can lead to different situation like:
```
this.setState({ name: "Zsolt" });
console.log(this.state.name); // this can log undefined
```
so with osage of callback
```
this.setState({ name: "Zsolt" }, () => console.log(this.state.name) ); // this will always log your name
```